src/mimir_coordinator.py

```diff
@@ -141,9 +149,20 @@ def _build_tls_config(self) -> Dict[str, Any]:
     # data_dir:
     # The Mimir Alertmanager stores the alerts state on local disk at the location configured using -alertmanager.storage.path.
     # Should be persisted if not replicated
+
+    # sharding_ring.replication_factor: int
+    # (advanced) The replication factor to use when sharding the alertmanager.
     def _build_alertmanager_config(self) -> Dict[str, Any]:
+        alertmanager_scale = len(
+            self._cluster_provider.gather_addresses_by_role()[MimirRole.alertmanager]
+        )
```

Comment:
As far as I understand, if `gather_addresses_by_role()` may return an empty dict. If that situation occurs, a `KeyError` exception will be raised.

```suggestion
        alertmanager_scale = len(
            self._cluster_provider.gather_addresses_by_role().get(MimirRole.alertmanager, None)
        )
```

Reply:
Done!